### PR TITLE
Implements Error for GetRecordError (#2612)

### PR DESCRIPTION
**Pull request by Janmajayamall**

### CODE REVIEW

**Review by elenaf9 — APPROVED** [review]
Thank you @Janmajayamall! Could you please add a changelog entry to `kad/CHANGELOG.md`?

**Comment by Janmajayamall** [comment]
yup done!

**Comment by elenaf9** [comment]
Not sure if it was intentional, but feca63f also changed some spacing and capitalization of older changelog entries. Mind reverting that?

**Comment by Janmajayamall** [comment]
Hey! Sorry didn't notice that. Just updated!

File: protocols/kad/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 - Update to `libp2p-swarm` `v0.36.0`.
 
+-   Derive `Error` for `kad::GetRecordError` (see [PR 2614])
```
Comment (mxinden) [inline protocols/kad/CHANGELOG.md]:
```suggestion
- Derive `Error` for `GetRecordError` (see [PR 2614]).
```

**Review by mxinden — APPROVED** [review]
Thanks @Janmajayamall!

**Pull request merged**